File: sdk/src/types/ledger.ts
import { z } from 'zod';

export enum LedgerEntryType {
  CREDIT = 'CREDIT',
  DEBIT = 'DEBIT',
}

export const LedgerEntrySchema = z.object({
  id: z.string(),
  tenantId: z.string(),
  transactionId: z.string(),
  type: z.nativeEnum(LedgerEntryType),
  amount: z.string(),
  currency: z.string(),
  balanceAfter: z.string(),
  referenceId: z.string().optional(),
  description: z.string().optional(),
  createdAt: z.string(),
});

export type LedgerEntry = z.infer<typeof LedgerEntrySchema>;

export const LedgerFilterSchema = z.object({
  transactionId: z.string().optional(),
  referenceId: z.string().optional(),
  startDate: z.string().optional(),
  endDate: z.string().optional(),
  limit: z.number().optional(),
  offset: z.number().optional(),
});

export type LedgerFilters = z.infer<typeof LedgerFilterSchema>;

File: sdk-python/src/rampos/models/ledger.py
"""Ledger models for transaction entries."""

from __future__ import annotations

from enum import Enum

from pydantic import BaseModel


class LedgerEntryType(str, Enum):
    CREDIT = "CREDIT"
    DEBIT = "DEBIT"


class LedgerEntry(BaseModel):
    id: str
    tenant_id: str
    transaction_id: str
    type: LedgerEntryType
    amount: str
    currency: str
    balance_after: str
    reference_id: str | None = None
    description: str | None = None
    created_at: str


class LedgerFilters(BaseModel):
    transaction_id: str | None = None
    reference_id: str | None = None
    start_date: str | None = None
    end_date: str | None = None
    limit: int | None = None
    offset: int | None = None
